### ISSUE TITLE
Providing score trend on plugins

### ISSUE THREAD

**Issue by alecharp** [jenkins-infra/plugin-health-scoring#493]
Knowing the current score of a plugin great, but knowing if it improves and is getting worse would be a nice addition to the data we have on each plugin.

In addition, knowing when that previous score was computed would help understand if the plugin is change frequently or not.

**Comment by AayushSaini101** [comment]
We already keep the last at max 5 scores in the scores table, We can return a list of the score to UI, then UI can decide if its shows the value or if it shows a sign for the trend
![image](https://github.com/jenkins-infra/plugin-health-scoring/assets/60972989/c4d41f2e-0d51-48b5-95fc-bef7066b86c3)

**Comment by AayushSaini101** [comment]
We just need to create a api to return a list of score with timestamp of generation, It would help the UI to analyse the trend easily

**Comment by alecharp** [comment]
No you don't.
Plugins Site is using one API endpoint and shouldn't have to multiply the calls.

In the current `/api/scores` output, we should add a new entry, for each plugin, to hold the previously known score.

**Comment by AayushSaini101** [comment]
> No you don't. Plugins Site is using one API endpoint and shouldn't have to multiply the calls.
> 
> In the current `/api/scores` output, we should add a new entry, for each plugin, to hold the previously known score.

Thanks @alecharp so the endpoint will be /api/scores/{pluginName} this will return all the score result

**Comment by alecharp** [comment]
No, that would still be a new endpoint. We want the trend to be accessible in the current `/api/scores` output.

**Comment by AayushSaini101** [comment]
> No, that would still be a new endpoint. We want the trend to be accessible in the current `/api/scores` 

@alecharp Thanks, in this case, we can create and pass another parameter in the score card,  `return bodyBuilder.body(new ScoreReport(plugins, scoreService.getScoresStatistics()));` that contains all the information of past score of a plugin

**Comment by alecharp** [comment]
No. We want the element of `plugins` to have a `previousScore` for example.

**Comment by alecharp** [comment]
Either you provide an array of previous score, or only one value. Last value is enough.

**Comment by AayushSaini101** [comment]
> Either you provide an array of previous score, or only one value. Last value is enough.

Thank you so much @alecharp I guess it is good to provide all previous score to better analysis

**Comment by alecharp** [comment]
No, only last score is better. It's easier to interpret.
